### ISSUE TITLE
 Incorrect property inheritance

### ISSUE THREAD

**Issue by jazzynico** [scour-project/scour#22]
(Original report: https://bugs.launchpad.net/scour/+bug/734019).

Properties seem to be inherited incorrectly in Scour. Here's the declaration in the example file gnome-dev-pcmcia.svg:

```SVG
  <g
    [...] id="text5059"
    style="[...];fill:none;stroke:black;stroke-width:0.99067909px;stroke-linecap:butt;stroke-linejoin:miter;stroke-opacity:1">
    <path
       [...] id="path3675"
       style="[...];fill:white;fill-opacity:1;stroke:none;[...]" />
```

But the path ends up with stroke:black due to incorrect inheritance rules. The text PCMCIA, described by the above-named path in the example file, is then black instead of white.

Scour version used = trunk r206
Command line used = scour.py --disable-style-to-xml -i INFILE -o OUTFILE

**Comment by jazzynico** [comment]
Reproduced on Xubuntu 15.10, Scour 0.31, with the following file:

```SVG
<?xml version="1.0" encoding="UTF-8" standalone="no"?>
<svg xmlns="http://www.w3.org/2000/svg" width="128" height="102">
  <g style="stroke:#000000;stroke-width:2.8">
    <path style="fill:#da121a" d="M 6.9543768,24.389491 C 61.495463,29.983045 56.419519,-9.1694984 126.76305,3.8352259 122.42037,29.071113 120.23284,54.307 119.3663,79.542887 71.957897,74.290215 47.722934,109.41712 1.2234113,98.22029 8.1828099,67.103233 0.71310807,52.681046 6.9543768,24.389491 z" />
    <path style="fill:#fcdd09;stroke:none" d="m 122.67878,21.181009 c -1.69605,12.613478 -2.9017,25.226931 -3.72117,37.840409 C 65.803602,52.464067 60.032853,85.797614 5.2209057,79.740429 6.3118538,65.717509 4.8645717,54.767235 5.7050061,41.647833 69.953417,49.142975 61.145206,10.107397 122.67878,21.181009 z" />
  </g>
</svg>
```

Result (scoured file):

```SVG
<?xml version="1.0" encoding="UTF-8" standalone="no"?>
<svg width="128" xmlns="http://www.w3.org/2000/svg" height="102">
 <g stroke-width="2.8" stroke="#000">
  <path d="m6.9544 24.389c54.541 5.594 49.466-33.558 119.81-20.554-4.34 25.236-6.53 50.472-7.39 75.708-47.412-5.253-71.647 29.877-118.15 18.677 6.9628-31.117-0.50689-45.539 5.7344-73.831z" fill="#da121a"/>
  <path d="m122.68 21.181c-1.696 12.613-2.9017 25.227-3.7212 37.84-53.156-6.557-58.927 26.777-113.74 20.719 1.0919-14.022-0.3554-24.973 0.485-38.092 64.248 7.495 55.44-31.541 116.98-20.467z" fill="#fcdd09"/>
 </g>
</svg>
```

**Comment by codedread** [comment]
Is this a regression?

**Comment by jazzynico** [comment]
@codedread - Apparently not. It was reported by Louis in 2011, and I confirmed it today.

**Comment by Ede123** [comment]
Issue is [here](https://github.com/scour-project/scour/blob/7c86de83d1b686c73adfef3ce241a921b2fb077a/scour/scour.py#L1399).

As Jeff already properly noted we have to check all parents for a differing definition of stroke before we can safely remove the `stroke="none"` from the node in question.

**Comment by Ede123** [comment]
Actually the other way round could be just as problematic:<br>If a parent defines `stroke="none"` but one of its children as a differing stroke specification, it might well inherit the other stroke attributes (which we currently remove) from the parent.

We definitely need a way to do a "getComputedStyle()" to solve this.

**Comment by Ede123** [comment]
Fixed in #111.
Please test and check if I overlooked any edge cases!

However there's still a lot of work to do in respect to the larger context of this bug:
* First of all we have to properly parse style sheets which might change the computed style of a node which we currently do *not* consider
* There's a lot of potential regarding inheritance rules. Right now Scour doesn't really know what inheritance is and the fixes above are only a start to really solve all issues of that sort.

All in all we should probably start thinking about implementation of a proper CSS parser and then find a way to efficiently implement a "getComputedStyle" function (but that's material for a new issue...).
